src/main/java/nwoolcan/model/brewery/production/batch/step/QueryParameter.java

```diff
@@ -0,0 +1,206 @@
+    private boolean sortDescending = false;
+
+    /**
+     * Empty constructor. In order to build the object use the builder methods.
+     */
+    public QueryParameter() { }
+
+    /**
+     * Returns the {@link ParameterType} to use as a filter in the query.
+     * @return the {@link ParameterType} to use as a filter in the query.
+     */
+    public Optional<ParameterType> getParameterType() {
+        return Optional.ofNullable(parameterType);
+    }
+
+    /**
+     * Returns the {@link Number} that represent the lower bound of the value filter.
+     * @return the {@link Number} that represent the lower bound of the value filter.
+     */
+    public Optional<Number> getGreaterThanValue() {
+        return Optional.ofNullable(greaterThanValue);
+    }
+
+    /**
+     * Returns the {@link Number} that represent the upper bound of the value filter.
+     * @return the {@link Number} that represent the upper bound of the value filter.
+     */
+    public Optional<Number> getLessThanValue() {
+        return Optional.ofNullable(lessThanValue);
+    }
+
+    /**
+     * Returns the {@link Number} that represent the exact value filter.
+     * @return the {@link Number} that represent the exact value filter.
+     */
+    public Optional<Number> getExactValue() {
+        return Optional.ofNullable(exactValue);
+    }
+
+    /**
+     * Returns the {@link Date} that represent the lower bound of the date filter.
+     * @return the {@link Date} that represent the lower bound of the date filter.
+     */
+    public Optional<Date> getStartDate() {
+        return Optional.ofNullable(startDate).map(d -> new Date(d.getTime()));
+    }
+
+    /**
+     * Returns the {@link Date} that represent the upper bound of the date filter.
+     * @return the {@link Date} that represent the upper bound of the date filter.
+     */
+    public Optional<Date> getEndDate() {
+        return Optional.ofNullable(endDate).map(d -> new Date(d.getTime()));
+    }
+
+    /**
+     * Returns true if needs to be sorted by value, false otherwise.
+     * @return true if needs to be sorted by value, false otherwise.
+     */
+    public boolean isSortByValue() {
+        return sortByValue;
+    }
+
+    /**
+     * Returns true if needs to be sorted by date, false otherwise.
+     * @return true if needs to be sorted by date, false otherwise.
+     */
+    public boolean isSortByDate() {
+        return sortByDate;
+    }
+
+    /**
+     * Returns true if sorting in descending order, false otherwise.
+     * @return true if sorting in descending order, false otherwise.
+     */
+    public boolean isSortDescending() {
+        return sortDescending;
+    }
+
+    /**
+     * Sets the parameter type value to filter.
+     * @param parameterType the parameter type value to filter.
+     * @return myself after setting the parameter type value to filter.
+     */
+    public QueryParameter parameterType(final ParameterType parameterType) {
```

Comment:
Single responsibility principle, should make an external class (or inner) to build QueryParameter

Reply:
Like a builder? I thought about it but it's an unneccessary repetition of code imho and a very large constructor

Reply:
Yep, why repetition of code? Just move the code that builds `QueryParameter` in a new class

Reply:
You need to repeat all fields in the builder

Reply:
That's how a builder works, i do not see any problem

Reply:
it is not useful since it is not a complex builder that needs to handle hard logic.
there is no need for classes that do not enhance code for the KISS principle.
does not break single responsibility principle since the class has only one responsibility : to build its self and nothing else.

Reply:
>the class has only one responsibility : to build its self and nothing else.

It also has to represent the `QueryParameter` object and related methods, still don't like it, @samueleceroni what do you think?

Reply:
resolved with #61

Reply:
i implemented this request